stdlib/public/core/ArraySlice.swift

```diff
@@ -0,0 +1,70 @@
+
+public typealias ArraySlice<T> = Slice<[T]>
+
+extension Slice: _ContiguouslyStored where Base: _ContiguouslyStored
+/* FIXME: exterminate */, Base.IndexDistance == Int 
+{
+  public func withUnsafeBufferPointer<R>(
+    _ body: (UnsafeBufferPointer<Element>) throws -> R
+  ) rethrows -> R {
+    let i = _base.distance(from: _base.startIndex, to: _startIndex)
+    let j = _base.distance(from: _base.startIndex, to: _endIndex)
+    return try _base.withUnsafeBufferPointer { buf in
+      let sliced = UnsafeBufferPointer(rebasing: buf[i..<j])
+      return try body(sliced)
+    }
+  }
+
+  public mutating func withUnsafeMutableBufferPointer<R>(
+    _ body: (inout UnsafeMutableBufferPointer<Element>) throws -> R
```

Comment:
I know this matches the existing API, but why is this `inout`? Mutating the buffer itself doesn't have any effect, does it?

Reply:
I have a feeling there is a good reason for this, which I can't recall.

(this is just placeholder code, I'm unsure yet whether there should be a `MutableContiguouslyStored` or something)

Reply:
@atrick will know.

Reply:
The `inout` pointer argument dates back to primordial Swift. Notice
that `withUnsafeMutableBytes`, added later, correctly passes a
by-value pointer. This is one of a number of source-breaking API
cleanups that I never got around to because nothing was blocked on it.